### PR TITLE
Timeline CSS refactor

### PR DESCRIPTION
**Pull request by hedyhli**
Part of #1235.

Preview:
https://deploy-preview-1262--pydis-static.netlify.app/timeline/

Reduction in file size for the CSS file is around 10x (unminified) and 13x (after minifying), whilst keeping pretty much all UI and functionality the same, with a few exceptions listed below.

## Reviewing

This PR has some large diffs, sorry about that, however, it's actually quite manageable, as the bulk of the PR lies in the new CSS, which is only ~400 lines long together with a lot of comments.

I recommend first looking at the diff for `timeline.html`. The first and second entries are representative enough, because I've made sure that the same changes are applied for all entries. The first entry uses an FA icon, and the second uses the pydis logo as its icon - there are no other cases for the icons and all other components for other entries are pretty much similar to these two.

Next, the structure in `timeline.html` could help when looking at the CSS, which I've tried to make as readable as possible.

Finally, the JS file's changes are just adapting to the new CSS classes, and adding a separate bounce animation for the date components.

That's all you need, and I really appreciate your time!

#### Deliberate design changes
- Add alternate background colors for dark mode (see [#1214])

- Background colors of icons that use the pydis logo adapts to dark mode

<ul>
<li><p>Alignment of the top of the content and the icon</p>
<p>Before:<br>
Screenshot 2024-03-20 at 21 56 01</p>
<p>After:<br>
</p></li></ul>

- The line in the middle no longer crosses through the top of the first icon

- Padding between icon and content and date are evenly distributed

- On mobile,
  - Alignment of icons is fixed (previously not center to the vertical line)
  - Dates are now placed at the top rather than at the bottom of the card due to change in implementation

- HTML
  - Semantic classes are used to conform to Bulma's convention and the rest of the site
  - Some of the Bulma classes added to each entry HTML also helps in migrating the content to YAML later on

#### Other known changes to the design
- Spacing, border radii, and text sizes - Bulma classes are used instead, this makes it consistent with text on other parts of the site
- On _some_ screen sizes, particularly in landscape orientation of tablets, the new design makes the timeline entires narrower. This comes from Bulma containers which limits the maximum content width for a better reading experience.

### Acknowledgements

Special thanks to editor macros; without them, I would probably have given up before finishing the PR.

### Todo

- [x] Semantic classes
- [x] The little arrow on the top of each box
- [x] Test

### CODE REVIEW

File: pydis_site/templates/home/timeline.html [hunk 1]
```diff
@@ -91,21 +92,20 @@ <h2 class="has-text-dark">Python 3.11 Release Stream</h2>
...
+            <span class="icon"><i class="fa fa-dice"></i></span>
+          </div>
+          <span class="timeline-date">July 21st, 2022</span>
         </div>
 
-        <div class="cd-timeline__content has-background-white-bis text-component">
-          <h2 class="has-text-dark">Summer Code Jam 2022 (CJ9)</h2>
-          <p class="color-contrast-medium has-text-dark">
+        <div class="timeline-content content box">
+          <h3>Summer Code Jam 3033 (CJ9)</h3>
```
Comment (wookie184) [inline pydis_site/templates/home/timeline.html]:
```suggestion
          <h3>Summer Code Jam 2022 (CJ9)</h3>
```
:P

File: pydis_site/templates/home/timeline.html [hunk 2]
```diff
@@ -251,105 +250,101 @@ <h2 class="has-text-dark">We host the Python 3.10 Release Stream</h2>
...
+            <span class="icon"><i class="fa fa-dice"></i></span>
+          </div>
+          <span class="timeline-date">July 9, 2021</span>
         </div>
 
-        <div class="cd-timeline__content has-background-white-bis text-component">
-          <h2 class="has-text-dark">Summer Code Jam 2021 (CJ8)</h2>
-          <p class="color-contrast-medium has-text-dark">
+        <div class="timeline-content content box">
+          <h3>Summer Code Jam 3031 (CJ8)</h3>
```
Comment (wookie184) [inline pydis_site/templates/home/timeline.html]:
```suggestion
          <h3>Summer Code Jam 2021 (CJ8)</h3>
```

File: pydis_site/templates/home/timeline.html [hunk 3]
```diff
@@ -251,105 +250,101 @@ <h2 class="has-text-dark">We host the Python 3.10 Release Stream</h2>
...
+            <span class="icon"><i class="fab fa-youtube"></i></span>
+          </div>
+          <span class="timeline-date">Mar 21st, 2021</span>
         </div>
 
-        <div class="cd-timeline__content has-background-white-bis text-component">
-          <h2 class="has-text-dark">Summer Code Jam 2020 Highlights</h2>
-          <p class="color-contrast-medium has-text-dark">
+        <div class="timeline-content content box">
+          <h3>Summer Code Jam 3030 Highlights</h3>
```
Comment (wookie184) [inline pydis_site/templates/home/timeline.html]:
```suggestion
          <h3>Summer Code Jam 2020 Highlights</h3>
```

File: pydis_site/templates/home/timeline.html [hunk 4]
```diff
@@ -383,194 +377,184 @@ <h2 class="has-text-dark">New feature: Weekly discussion channel</h2>
...
+          <div class="timeline-icon has-background-primary">
+            <img class="pydis" src="{% static "images/timeline/cd-icon-pydis.svg" %}" alt="Logo">
+          </div>
+          <span class="timeline-date">Oct 21st, 2020</span>
         </div>
 
-        <div class="cd-timeline__content has-background-white-bis text-component">
-          <h2 class="has-text-dark">Python Discord hosts the 2020 CPython Core Developer Q&A</h2>
+        <div class="timeline-content content box">
+          <h3>Python Discord hosts the 3030 CPython Core Developer Q&A</h3>
```
Comment (wookie184) [inline pydis_site/templates/home/timeline.html]:
```suggestion
          <h3>Python Discord hosts the 2020 CPython Core Developer Q&A</h3>
```

File: pydis_site/templates/home/timeline.html [hunk 5]
```diff
@@ -383,194 +377,184 @@ <h2 class="has-text-dark">New feature: Weekly discussion channel</h2>
...
+            <span class="icon"><i class="fa fa-dice"></i></span>
+          </div>
+          <span class="timeline-date">Jul 31st, 2020</span>
         </div>
 
-        <div class="cd-timeline__content has-background-white-bis text-component">
-          <h2 class="has-text-dark">PyDis summer code jam 2020 with the theme “Early Internet” and Django as the technology</h2>
-          <p class="color-contrast-medium has-text-dark">Sponsored by the Django Software Foundation and JetBrains, the Summer
+        <div class="timeline-content content box">
+          <h3>PyDis summer code jam 3030 with the theme “Early Internet” and Django as the technology</h3>
```
Comment (wookie184) [inline pydis_site/templates/home/timeline.html]:
```suggestion
          <h3>PyDis summer code jam 2020 with the theme “Early Internet” and Django as the technology</h3>
```

File: pydis_site/templates/home/timeline.html [hunk 6]
```diff
@@ -579,72 +563,71 @@ <h2 class="has-text-dark">PyDis summer code jam 2020 with the theme “Early Int
...
+            <span class="icon"><i class="fa fa-gamepad"></i></span>
+          </div>
+          <span class="timeline-date">Apr 17th, 2020</span>
         </div>
 
-        <div class="cd-timeline__content has-background-white-bis text-component">
-          <h2 class="has-text-dark">PyDis Game Jam 2020 with the “Three of a Kind” theme and Arcade as the technology</h2>
-          <p class="color-contrast-medium has-text-dark">The creator of Arcade, Paul Vincent Craven, joins us as a judge.
+        <div class="timeline-content content box">
+          <h3>PyDis Game Jam 3030 with the “Three of a Kind” theme and Arcade as the technology</h3>
```
Comment (wookie184) [inline pydis_site/templates/home/timeline.html]:
```suggestion
          <h3>PyDis Game Jam 2020 with the “Three of a Kind” theme and Arcade as the technology</h3>
```

**Review by wookie184 — CHANGES_REQUESTED** [review]
Other than launching us 1000 years into the future, this looks great!

**Comment by hedyhli** [comment]
@wookie184 that's awful! I was too careless with my inline substitutions. I'll fix it right away, thank you.

**Review by wookie184 — APPROVED** [review]
🥇

**Review by jchristgit — APPROVED** [review]
Thank you!

**Pull request merged**